server/game/cards/02_SHD/units/ChainCodeCollector.ts

```diff
@@ -0,0 +1,32 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Duration, KeywordName } from '../../../core/Constants';
+
+export default class ChainCodeCollector extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '7171636330',
+            internalName: 'chain-code-collector'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addOnAttackAbility({
+            title: 'If the defender has a Bounty, it gets –4/–0 for this attack',
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
+                condition: (context) => {
+                    const target = context.event.attack.target;
+                    return target.isUnit() && target.hasSomeKeyword(KeywordName.Bounty);
+                },
+                onTrue: AbilityHelper.immediateEffects.cardLastingEffect((context) => ({
```

Comment:
I think it would be good to have a `cardAttackLastingEffectSystem` similar to `cardPhaseLastingEffectSystem` but it will require some mild refactoring (very easy). Would you mind just creating a copy of CardPhaseLastingEffectSystem.ts named CardAttackLastingEffectSystem.ts (and then add a new function for it in GameSystemLibrary named `forThisAttackCardEffect`).

Other than renaming stuff, the only change needed would be to add this function:

```typescript
public override canAffect(card: Card, context: TContext) {
    if (!card.isUnit()) {
        return false;
    }

    Contract.assertNotNullLike(card.activeAttack, `Attempting to apply an attack lasting effect to ${card.internalName} but it is not actively attacking`);

    return super.canAffect(card, context);
}
```